tuned_lens/model_surgery.py

```diff
@@ -221,6 +221,87 @@ def replace_layers(
 
     set_key_path_(model, list_path, modified_list)
     try:
-        yield model
+        yield
     finally:
         set_key_path_(model, list_path, layer_list)
+
+
+class _AddVector(th.nn.Module):
+    def __init__(
+        self,
+        base: th.nn.Module,
+        vector: th.Tensor,
+        get_hidden: Callable[[Any], th.Tensor],
+        set_hidden: Callable[[Any, th.Tensor], Any],
+    ):
+        super().__init__()
+        self.base = base
+        self.vector = nn.Parameter(vector.clone())
+        self.get_hidden = get_hidden
+        self.set_hidden = set_hidden
+
+    def forward(self, *args, **kwargs) -> th.Tensor:
+        outputs = self.base(*args, **kwargs)
+        hidden = self.get_hidden(outputs)
+        hidden = hidden + self.vector
+        outputs = self.set_hidden(outputs, hidden)
+        return outputs
+
+
+@contextmanager
+def add_steering_vector(
+    model: Model,
+    indices: list[int],
```

Comment:
What happens if indices are repeated? As I understand it, the last value assigned to an index is picked (due to how `replace_layers` is written), but this should be asserted out or documented.

Reply:
Yeah we should probably add a `assert len(set(indices)) == len(indices)` check or something